add-chain/addresses.go

```diff
@@ -142,68 +142,94 @@ func readAddressesFromJSON(contractAddresses map[string]string, deploymentsDir s
+		return fmt.Errorf("invalid deployment filepath provided: %w", err)
+	}
 
-		_, err := os.ReadFile(filepath.Join(deploymentsDir, AnchorStateRegistryProxy+".json"))
-		if errors.Is(err, os.ErrNotExist) {
-			contracts = contractsFromJSONNonFaultProofs
-		}
-		for _, name := range contracts {
-			path := filepath.Join(deploymentsDir, name+".json")
-			file, err := os.ReadFile(path)
-			if err != nil {
-				return fmt.Errorf("failed to read file: %w", err)
+	if fileInfo.IsDir() {
+		deployFilePath = filepath.Join(deploymentsDir, ".deploy")
+		_, err = os.Stat(deployFilePath)
+		if err != nil {
+			// Use legacy deployment artifact schema
+			fmt.Printf("failed to find .deploy file. Will look for legacy .json files")
+
+			if _, err := os.ReadFile(filepath.Join(deploymentsDir, FaultDisputeGame+".json")); errors.Is(err, os.ErrNotExist) {
+				contracts = append(contracts, faultProofContracts...)
+			} else if _, err := os.ReadFile(filepath.Join(deploymentsDir, AnchorStateRegistryProxy+".json")); errors.Is(err, os.ErrNotExist) {
+				contracts = append(contracts, customGasTokenContracts...)
+			} else {
+				contracts = append(contracts, L2OutputOracleProxy)
 			}
-			var data AddressData
-			if err = json.Unmarshal(file, &data); err != nil {
-				return fmt.Errorf("failed to unmarshal json: %w", err)
+
+			for _, name := range contracts {
+				path := filepath.Join(deploymentsDir, name+".json")
+				file, err := os.ReadFile(path)
+				if err != nil {
+					return fmt.Errorf("failed to read file: %w", err)
+				}
+				var data AddressData
+				if err = json.Unmarshal(file, &data); err != nil {
+					return fmt.Errorf("failed to unmarshal json: %w", err)
+				}
+				contractAddresses[name] = data.Address
 			}
-			contractAddresses[name] = data.Address
-		}
-	} else {
-		var addressList superchain.AddressList
-		rawData, err := os.ReadFile(deployFilePath)
-		if err != nil {
-			return fmt.Errorf("failed to read file: %w", err)
+			return nil
 		}
+	}
 
-		if err = json.Unmarshal(rawData, &addressList); err != nil {
-			return fmt.Errorf("failed to unmarshal json: %w", err)
-		}
+	var addressList superchain.AddressList
+	rawData, err := os.ReadFile(deployFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read file: %w", err)
+	}
 
-		_, err = addressList.AddressFor(AnchorStateRegistryProxy)
-		if err != nil {
-			contracts = contractsFromJSONNonFaultProofs
-		}
+	if err = json.Unmarshal(rawData, &addressList); err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
 
-		for _, name := range contracts {
-			address, err := addressList.AddressFor(name)
-			if err != nil {
-				return fmt.Errorf("failed to retrieve %s address from list: %w", name, err)
-			}
-			contractAddresses[name] = address.String()
+	if _, err = addressList.AddressFor(FaultDisputeGame); err == nil {
+		contracts = append(contracts, faultProofContracts...)
+	} else if _, err = addressList.AddressFor(AnchorStateRegistryProxy); err == nil {
```

Comment:
so FDG and CGT are mutually exclusive?

Reply:
at the moment, once https://github.com/ethereum-optimism/optimism/pull/10780 is merged then we will support cgt with fp